prime-router/src/main/kotlin/history/SubmissionHistory.kt

```diff
@@ -529,10 +539,10 @@ class DetailedSubmissionHistory(
              * Note: This method only works for the universal pipeline as the covid pipeline does the filtering and
              * routing in one step.
              */
-            if (actionsPerformed == setOf(TaskAction.receive)) {
-                return Status.RECEIVED
+            if (actionsPerformed.contains(TaskAction.route) && itemCountAfterFiltering == 0) {
+                return Status.NOT_DELIVERING
```

Comment:
Maybe this should be "waiting to deliver" if `itemCountAfterFiltering` is greater than 0?

Reply:
The acceptance criteria actually does say "received" so I'm fine with this either way!